index.js

```diff
@@ -396,28 +423,43 @@ SendStream.prototype.isRangeFresh = function isRangeFresh(){
  * @private
  */
 
-SendStream.prototype.redirect = function redirect(path) {
+SendStream.prototype.redirect = function redirectDirectory(path) {
   if (listenerCount(this, 'directory') !== 0) {
     this.emit('directory')
     return
   }
 
   if (this.hasTrailingSlash()) {
-    this.error(403)
-    return
+    return this.error(403)
   }
 
-  var loc = path + '/'
-  var msg = 'Redirecting to <a href="' + escapeHtml(loc) + '">' + escapeHtml(loc) + '</a>\n'
-  var res = this.res
+  redirect(this.res, path + '/')
+}
 
-  // redirect
-  res.statusCode = 301
-  res.setHeader('Content-Type', 'text/html; charset=UTF-8')
-  res.setHeader('Content-Length', Buffer.byteLength(msg))
-  res.setHeader('X-Content-Type-Options', 'nosniff')
-  res.setHeader('Location', loc)
-  res.end(msg)
+/**
+ * Redirect a symbolic link.
+ *
+ * @param {string} path
+ * @private
+ */
+
+SendStream.prototype.redirectSymbolicLink = function redirectSymbolicLink(path) {
+  var self = this
+
+  fs.readlink(path, function(err, linkString)
+  {
+    if (err) return self.onStatError(err)
+
+    // Get relative paths for all symlinks, also for absolute ones
+    path       = dirname(path)
+    linkString = relative(path, resolve(path, linkString))
```

Comment:
I would say, as a default, a 403 (option 2) may be the best. Option 3 is not great, because it means this feature is no longer "redirect on symlinks" but instead "maybe redirect on symlinks, maybe not; it depends on what your root is, ultimately". Option 1 is probably not a good option to give.